apps/data-pipeline/library-traffic-scraper/src/lib.ts

```diff
@@ -0,0 +1,107 @@
+import { database } from "@packages/db";
+import { libraryTraffic, libraryTrafficHistory } from "@packages/db/schema";
+import fetch from "cross-fetch";
+
+type RawRespOK = {
+  message: "OK";
+  data: {
+    id: number;
+    name: string;
+    count: number;
+    percentage: number;
+    timestamp: string;
+  };
+};
+
+type RawRespErr = { error: string };
+
+async function fetchLocation(id: string): Promise<RawRespOK["data"] | null> {
+  const url = `https://www.lib.uci.edu/sites/all/scripts/occuspace.php?id=${id}`;
+  try {
+    const raw = await fetch(url).then((r) => r.text());
+
+    const parsedOnce = JSON.parse(raw);
+    const parsed = typeof parsedOnce === "string" ? JSON.parse(parsedOnce) : parsedOnce;
+
+    if (parsed.data && typeof parsed.data === "object") {
+      return (parsed as RawRespOK).data;
+    }
+
+    console.warn(`Location ID ${id} returned error: ${(parsed as RawRespErr).error}`);
+  } catch (err) {
+    console.error(`Unexpected error while fetching location ID ${id}:`, err);
+    throw err;
+  }
+  return null;
+}
+
+export async function doScrape(DB_URL: string): Promise<void> {
+  console.log("Starting library traffic scrape.");
+
+  const pageHtml = await fetch("https://www.lib.uci.edu/where-do-you-want-study-today").then((r) =>
+    r.text(),
+  );
+
+  const scriptBlockMatch = pageHtml.match(
+    /<script\b[^>]*>([\s\S]*?locationIds[\s\S]*?)<\/script>/i,
```

Comment:
is there a good reason to not use `cheerio` here? it would probably make your life drastically easier and make this code more readable